corehq/apps/importer/util.py

```diff
@@ -404,12 +410,16 @@ def populate_updated_fields(config, columns, row, datemode):
 
         if update_value is not None:
             if field_map[key]['type_field'] == 'date':
-                update_value = parse_excel_date(update_value, datemode)
-            elif field_map[key]['type_field'] == 'integer':
                 try:
-                    update_value = str(int(update_value))
-                except ValueError:
-                    update_value = ''
+                    update_value = parse_excel_date(update_value, datemode)
+                except InvalidDateException:
```

Comment:
Is this sketchy? I think the other way is to not catch the error in `parse_excel_date` and just catch it here.